### ISSUE TITLE
Matrix rank() produces wrong result

### ISSUE THREAD

**Issue by JayHoe** [sympy/sympy#11434]
Consider the following code:

``` python
ax,ay,bx,by,cx,cy,dx,dy,ex,ey,t0,t1 = symbols('a_x a_y b_x b_y c_x c_y d_x d_y e_x e_y t_0 t_1')
M=Matrix([[ax,ay,ax*t0,ay*t0,0],[bx,by,bx*t0,by*t0,0],[cx,cy,cx*t0,cy*t0,1],[dx,dy,dx*t0,dy*t0,1],[ex,ey,2*ex*t1-ex*t0,2*ey*t1-ey*t0,0]])
```

M.rank()
5

M.det()
0

You cannot have full rank of 5 and zero determinant. Cross-checked it on wolframalpha.com which gives 4 for the matrix rank. Thus, the rank calculation must be wrong.

**Comment by moorepants** [comment]
We have an issue open for these types of errors here: https://github.com/sympy/sympy/issues/9480

**Comment by JayHoe** [comment]
M.rank(simplify=True) works as expected and yields 4 as matrix rank.

**Comment by siefkenj** [comment]
This should be resolved by PR #11554.  @JayHoe  can you verify this and close the issue if true?

**Comment by moorepants** [comment]
Tested here:

```

In [1]: from sympy import *

In [2]: %paste
ax,ay,bx,by,cx,cy,dx,dy,ex,ey,t0,t1 = symbols('a_x a_y b_x b_y c_x c_y d_x d_y e_x e_y t_0 t_1')
M=Matrix([[ax,ay,ax*t0,ay*t0,0],[bx,by,bx*t0,by*t0,0],[cx,cy,cx*t0,cy*t0,1],[dx,dy,dx*t0,dy*t0,1],[ex,ey,2*ex*t1-ex*t0,2*ey*t1-ey*t0,0]])

## -- End pasted text --

In [3]: M.rank()
Out[3]: 4
```
